### DISCUSSION TITLE
Inquiry about useSlots functionality in GitHub Primer

### DISCUSSION THREAD

**Discussion by RebecaLiz** [primer/react#4570] [General] [answered]
Dear GitHub Support Team,

I hope this message finds you well.

I've been exploring the GitHub Primer UI system and came across the useSlots functionality in some components. I find this feature intriguing and would like to learn more about its implementation and usage within Primer.

Could you kindly provide me with more information about useSlots? Specifically, I'm interested in understanding how it functions within the Primer system, any relevant documentation, and perhaps if there's an RFC (Request for Comments) or similar resource detailing its design and purpose.

Your assistance in clarifying this matter would be greatly appreciated.

Thank you for your time and attention.

Best regards,

**Comment by lesliecdubs** [comment, marked as answer]
Hi there! Primer's usage of slots tends to be internal-facing so there's no public API or docs. You can check out the tests https://github.com/primer/react/blob/main/packages/react/src/hooks/__tests__/useSlots.test.tsx#L98 which might offer some insight.
